src/main/java/core/basesyntax/dao/ActivityDao.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Activity;
+import java.util.List;
+
+public interface ActivityDao {
+    List<Activity> getAll();
```

Comment:
```suggestion
public interface FruitRecordParser {
    List<FruitRecord> parse(List<String> lines);
```

please don't make it depend on the reader
in your `main` method call reader, get lines and then pass them to parser

Reply:
Sorry but I don't quite get this point. I have FruitRecordDao whose job is to perform CRUD operations on FruitRecords. And then I have FruitRecordDaoCsvFileImpl that implements the behaviour of getting FruitRecords from CSV File. If I will have another provider of data such as DB, JSON, XML or another file I will just add another implementation of my FruitRecordDao and everything will be working the same way

Reply:
we have discussed this during the QnA

our services should be independent
reader should not know about FruitRecordParser
FruitRecordParser should not know about the source (csv file, some other file, or other source), and it should not know about reader
same for FruitService and parser, fruitService should accept the `ready-to-use` data